workflow/snakemake_rules/main_workflow.smk

```diff
@@ -1033,7 +1033,7 @@ rule colors:
         # Memory use scales primarily with the size of the metadata file.
         # Compared to other rules, this rule loads metadata as a pandas
         # DataFrame instead of a dictionary, so it uses much less memory.
-        mem_mb=lambda wildcards, input: 5 * int(input.metadata.size / 1024 / 1024)
+        mem_mb=lambda wildcards, input: 5 * input.size_files_mb[next(iter([i for i,x in enumerate(input.keys()) if x=='metadata']))]
```

Comment:
`input.size_files_mb[list(input.keys()).index('metadata')]` is simpler, but we should spend some more time poking around the `snakemake.io.InputFiles` class as this feels overly complex